doc/source/dev/create_release.rst

```diff
@@ -1,3 +1,14 @@
+Creating Galaxy Releases
+========================
+
+The "main" release process is an interactive checklist with instructions (see `Publication of Galaxy Release v 23.2 https://github.com/galaxyproject/galaxy/issues/16742>`_ for an example).
+This issue is generated via `make release-issue` at the end of the process.
+The final result of the release process are
+- a new branch (release_YY.N) from which point releases are created
+- a tag pointing at the first commit of the branch (vYY.N)
```

Comment:
Not sure if we still want this ? It doesn't hurt anything, but maybe we should let the first point release create the first tag ?